Library/Homebrew/download_strategy.rb

```diff
@@ -269,6 +269,17 @@ def initialize(name, resource)
 
   def fetch
     ohai "Downloading #{@url}"
+
+    url = resolve_url(@url)
+    if url != @url
+      if @url.start_with?("https://") && url.start_with?("http://") &&
+         !ENV["HOMEBREW_NO_INSECURE_REDIRECT"].nil?
+         raise "Insecure redirect detected! Escaping to safety."
```

Comment:
`https to http redirect detected and HOMEBREW_NO_INSECURE_REDIRECT is set so aborting`